### PR TITLE
streams.part1.exercise2 (Aliberdov)

### PR DESCRIPTION
**Pull request by timaliberdov**

### CODE REVIEW

File: src/test/java/streams/part1/exercise/Exercise2.java [hunk 1]
```diff
@@ -16,7 +18,11 @@
     public void calcAverageAgeOfEmployees() {
         List<Employee> employees = Example1.getEmployees();
 
-        Double expected = null;
+        Double expected = employees.stream()
+            .map(Employee::getPerson)
+            .mapToDouble(Person::getAge)
+            .average()
+            .orElse(0.0D);
```
Comment (AlexLuk) [inline src/test/java/streams/part1/exercise/Exercise2.java]: orElseThrow is more convenient here

File: src/test/java/streams/part1/exercise/Exercise2.java [hunk 2]
```diff
@@ -25,7 +31,10 @@ public void calcAverageAgeOfEmployees() {
     public void findPersonWithLongestFullName() {
         List<Employee> employees = Example1.getEmployees();
 
-        Person expected = null;
+        Person expected = employees.stream()
+            .map(Employee::getPerson)
+            .max(Comparator.comparingInt(p -> p.getFullName().length()))
+            .get();
```
Comment (AlexLuk) [inline src/test/java/streams/part1/exercise/Exercise2.java]: orElseThrow is more convenient here

File: src/test/java/streams/part1/exercise/Exercise2.java [hunk 3]
```diff
@@ -34,7 +43,15 @@ public void findPersonWithLongestFullName() {
...
-        Employee expected = null;
+        Employee expected = employees.stream()
+            .max(
+                Comparator.comparingInt(
+                    e -> e.getJobHistory().stream()
+                        .max(Comparator.comparingInt(JobHistoryEntry::getDuration))
+                        .map(JobHistoryEntry::getDuration).orElse(0)
+                )
+            )
+            .get();
```
Comment (AlexLuk) [inline src/test/java/streams/part1/exercise/Exercise2.java]: orElseThrow is more convenient here

**Review by AlexLuk — CHANGES_REQUESTED** [review]

File: src/test/java/streams/part1/exercise/Exercise2.java (hunk 3 above)
Comment (timaliberdov) [reply src/test/java/streams/part1/exercise/Exercise2.java]: Replaced get() with orElseThrow()

**Review by AnastasiyaSoloveva — APPROVED** [review]

**Review by Astel — APPROVED** [review]

**Pull request closed without merging**